skopt/acquisition.py

```diff
@@ -296,3 +305,54 @@ def gaussian_ei(X, model, y_opt=0.0, xi=0.01, return_grad=False):
         return values, grad
 
     return values
+
+
+def gaussian_mi(X, model, y_opt=0.0, alpha=1.6, gamma=0.0, return_grad=False):
+    """
+    Use mutual information to calculate the acquisition values
+
+    Parameters
+    ----------
+    * `X` [array-like, shape=(n_samples, n_features)]:
+        Values where the acquisition function should be computed.
+
+    * `model` [sklearn estimator that implements predict with ``return_std``]:
+        The fit estimator that approximates the function through the
+        method ``predict``.
+        It should have a ``return_std`` parameter that returns the standard
+        deviation.
+
+    * `y_opt` [float, default 0]:
+        Previous minimum value which we would like to improve upon.
+
+    * `gamma`: [float, default=0.0]:
+        Controls the exploration of the algorithm.
+
+    * `alpha`: [float, default=1.6]:
+        Governs the trade-off between precision and confidence.
+        Defined as alpha = log(1/delta) for 0 < delta < 1.
+        Bounds the cumulative regret with confidence 1-delta.
+        Must fall within the interval [0.30, 2.30].
+
+    Returns
+    -------
+    * `values`: [array-like, shape=(X.shape[0],)]:
+        Acquisition function values computed at X.
+    """
+    if return_grad:
+        mu, std, mu_grad, std_grad = model.predict(
+            X, return_std=True, return_mean_grad=True,
+            return_std_grad=True)
+    else:
+        mu, std = model.predict(X, return_std=True)
+
+    values = np.zeros_like(mu)
+    phi = np.sqrt(alpha) * (np.sqrt(std + gamma) - np.sqrt(gamma))
+    values = mu + phi
+    gamma = np.max(gamma + std)
```

Comment:
I agree, that is one of the cleanest solutions, since we have to update the value anyway in the optimizer.